File: Easy/Sqrt(x).py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Aug  2 12:37:51 2022

@author: jamescrisp
"""

class Solution:
    def mySqrt(self, x: int) -> int:
        
        # Find aproximate square-root
        
        prev = 0
        nex = 0
        over = False
        
        while over == False:
            nex = prev + 1
            if nex*nex > x:
                return prev
            prev = nex
        
        return False
                
        
        

File: Easy/Sqrt(x).java
/**
 * Return the square root of x rounded down to the nearest integer.
 * 
 * @author James Crisp
 */
class Solution {
    public int mySqrt(int x) {
        for (int i = 0; i <= x + 1; i++) {
            if (i * i > x) {
                return i - 1;
            }
        }
        return -1;
    }
}